### PR TITLE
Features

### PR DESCRIPTION
**Pull request by EliseJane**
Hi, team glamorous raven!

We created a custom "complete" action and route, so please look at that.
There is a database seed file for your convenience. After running rake db:migrate, run rake db:seed
We used several view helpers and partials, in case you don't see where the code is.
We 'borrowed' the flash messages code from the postit template.
Please ignore any bootstrap styling.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -39,6 +39,7 @@ group :development, :test do
   # Adds support for Capybara system testing and selenium driver
```
Comment (EliseJane) [inline Gemfile]: practicing leaving comments

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  root to: 'todos#index'
+
+  resources :todos
+
+  post '/todos/:id/complete', to: 'todos#complete'
```
Comment (p-reznick) [inline config/routes.rb]: Good job with the custom route -- our approach was very similar!

File: app/controllers/todos_controller.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
+class TodosController < ApplicationController
+  before_action :set_todo, except: [:new, :index, :create]
+
+  def index
+    @complete = Todo.where(completed: true).sort_by &:duedate
+    @incomplete = Todo.where(completed: false).sort_by &:duedate
```
Comment (p-reznick) [inline app/controllers/todos_controller.rb]: Interesting and intuitive approach splitting the list this way, we ended up taking a two-parameter sort approach.

File: app/controllers/todos_controller.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  end
+
+  def new
+    @todo = Todo.new
+    @todo.duedate = Time.now
+  end
+
+  def create
+    @todo = Todo.new(todo_params)
+    @todo.duedate = Time.new(params[:year], params[:month], params[:day])
```
Comment (p-reznick) [inline app/controllers/todos_controller.rb:19]: Good idea to use the `Time` class to integrate the params this way, simplifies the implementation of the form in the template.

File: app/views/todos/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<%= render 'errors', obj: @todo %>
+
+<div class="span8 well">
```
Comment (p-reznick) [inline app/views/todos/_form.html.erb:3]: Great idea to reduce duplication with a form partial!

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,13 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
```
Comment (p-reznick) [inline db/seeds.rb:8]: Thanks for the seed file :)

File: app/controllers/todos_controller.rb [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
```
Comment (preeV42500) [inline app/controllers/todos_controller.rb:20]: I like the use of Time.new with the params for creating the due date :)

**Comment by vedala** [comment]
- Great, intuitive solution for separating year, month and day.
- Smart helper `completed_todo_style`
- Good use of partials.
- Incomplete and complete todo items handled very nicely.

Overall an excellent solution!

File: db/seeds.rb (hunk 6 above)
Comment (sunny-b) [reply db/seeds.rb:8]: That was all Nitin

File: app/controllers/todos_controller.rb (hunk 3 above)
Comment (EliseJane) [reply app/controllers/todos_controller.rb]: I didn't know we could sort by two parameters. I'll look in to that. That way sounds better.

File: app/helpers/application_helper.rb [hunk 8]
```diff
@@ -1,2 +1,9 @@
 module ApplicationHelper
+  def format_date(dt)
+    dt.strftime("%b %e, %Y")
+  end
+
```
Comment (preeV42500) [inline app/helpers/application_helper.rb:5]: Nice job with having a helper method to format the date for the user interface!

File: app/controllers/todos_controller.rb (hunk 3 above)
Comment (knwang) [reply app/controllers/todos_controller.rb]: Todo.where(completed: true).order(:duedate)

**Pull request merged**